components/layout/fragment.rs

```diff
@@ -882,16 +882,26 @@ impl Fragment {
         let size = LogicalSize::new(self.style.writing_mode,
                                     split.inline_size,
                                     self.border_box.size.block);
-        let flags = match self.specific {
-            SpecificFragmentInfo::ScannedText(ref info) => info.flags,
-            _ => ScannedTextFlags::empty()
+        let (flags, insertion_point) = match self.specific {
+            SpecificFragmentInfo::ScannedText(ref info) => {
+                match info.insertion_point {
+                    Some(index) => {
+                        if split.range.contains(index) {
+                            (info.flags, info.insertion_point)
+                        } else {
+                            (info.flags, None)
+                        }
+                    }
```

Comment:
nit: could use a guard here and remove a level of indentation: `Some(index) if split.range.contains(index) => (info.flags, info.insertion_point), Some(_) =>(info.flags, None),`